metadataserver/inject/inject_linux.go

```diff
@@ -326,6 +328,30 @@ done:
 		return nil, fmt.Errorf("Could not bind listener socket: %w", err)
 	}
 
+	err = containerNSHandle.NeighSet(&netlink.Neigh{
+		LinkIndex:    metadataservice.Attrs().Index,
+		Family:       netlink.FAMILY_V6,
+		State:        netlink.NUD_PERMANENT,
+		Type:         unix.RTN_UNICAST,
+		IP:           tocontainerLinkLocal.IP,
+		HardwareAddr: tocontainer.Attrs().HardwareAddr,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("Cannot insert neighbor entry %w from container NS to imds NS", err)
+	}
+
+	err = intermediateNSHandle.NeighSet(&netlink.Neigh{
+		LinkIndex:    tocontainer.Attrs().Index,
+		Family:       netlink.FAMILY_V6,
+		State:        netlink.NUD_PERMANENT,
+		Type:         unix.RTN_UNICAST,
+		IP:           metadataserviceLinkLocal.IP,
+		HardwareAddr: metadataservice.Attrs().HardwareAddr,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("Cannot insert neighbor entry %w from imds NS to container NS", err)
+	}
```

Comment:
I look forward to when we can do this same trick for the ipv6 router or gateway or whatever.

Reply:
You mean on eth0? Or...?

Reply:
Yes, something like:

```
containerNSHandle.NeighSet(&netlink.Neigh{
  LinkIndex:    eth0.Attrs().Index,	
  Family:       netlink.FAMILY_V6,
  State:        netlink.NUD_PERMANENT,	
  Type:         unix.RTN_UNICAST,	
  IP:           ourIPv6GatewayIP.IP,	
  HardwareAddr: whateverOurIPv6RouterIsStaticallyGivenToUsFromEC2.Attrs().HardwareAddr,
}
```

And then never deal with neighbor discovery again?

Reply:
Getting the actual MAC of the phantom router is non-trivial :(.